### PR TITLE
[New Extension] Jump

### PR DESCRIPTION
**Pull request by SKaplanOfficial**
## Description

This extension implements a jump shortcut feature in Raycast, mimicking and expanding on the functionalities of command-line tools such as [Zsh-z](https://github.com/agkozak/zsh-z). Jump allows you to quickly navigate to files, folders, applications, and websites using abbreviations. Jump uses weighting logic to allow for very shorthand abbreviations for destinations that you jump to often.

### Available Commands

- Jump To Destination
  - Requires a _destination_ parameter.
  - Attempts to jump to the desired destination.
  - Allows for direct jumps to exact paths and URLs.
- Copy Jump Data
  - Copies a JSON representation of all jump destinations and associated weights to the clipboard.
- Import Jump Data
  - Imports destinations and weights from a JSON string.

### Examples

All of the following can* jump to google.com:
- `Jump To Destination [https://google.com]`
- `Jump To [google.com]`
- `Jump [goog]`
- `j [go]`

All of the following can* jump to Calculator.app:
- `Jump To Destination [/System/Applications/Calculator.app]`
- `Jump To [Calculator.app]`
- `Jump [calc]`
- `j [c]`

All of the following can* jump to the user's Downloads folder:
- `Jump To Destination [/Users/exampleUser/Downloads]`
- `Jump To [~/Downloads]`
- `Jump [downl]`
- `j [d]`

* = The actual destination resulting from shorthand abbreviations depends on the associated weights; as users use Jump more, their abbreviations become more specialized to their usage habits.

### Unimplemented Features / Future Direction
- Weights do not currently account for the recency of the last jump. This will be added in a future version.
- Weights do not currently account for the average time of day a jump shortcut is used. This would improve jump accuracy in different contexts (such as at work vs. at home) and will be explored for future versions.
- Jumps do not currently account for the proximity of the target, e.g. when attempting to jump to a subfolder of the currently open Finder window, that subfolder should be preferred over a more distant (non-child) folder. This will be explored for future versions.

## Screencast

[Jump.webm](https://user-images.githubusercontent.com/7865925/213685905-716579c2-aef2-4aea-b9ea-cb57a36bc1bf.webm)

[YouTube Video](https://www.youtube.com/watch?v=Ffgc6URsX98)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/jump/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Jump
+
+Jump to websites, folders, or files
```
Comment (pernielsentikaer) [inline extensions/jump/README.md:3]: I think most of the users (including myself) would struggle to understand how it works, I saw your demo, and it works as intended, but where do we get the data from, and how to generate it? 

Might be good to describe it in the README so it's possible to use 😊

Comment (SKaplanOfficial) [reply extensions/jump/README.md:3]: I've updated the README to describe the functionality better, what do you think of it now?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Thanks for all the update @SKaplanOfficial 🔥 

If you try to send malformed or empty data to `Import Jump Data` it just crashes, can we somehow catch this gracefully?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Thanks for the fix; if you just send the form without doing anything, it's still possible to crash the extension

https://user-images.githubusercontent.com/38033107/215990757-ff65fcfe-c6fa-4e1d-ae46-837962afc323.mp4

You can maybe use [useForm](https://developers.raycast.com/utilities/react-hooks/useform) but not required 😊

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋

Approved; let's 🚢 it!

**Pull request merged**